test/e2e/infoblox.go

```diff
@@ -0,0 +1,296 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+
+	ibclient "github.com/infobloxopen/infoblox-go-client"
+
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
+)
+
+const (
+	infobloxGridConfigDirEnvVar = "INFOBLOX_CONFIG_DIR"
+	infobloxGridHostEnvVar      = "INFOBLOX_GRID_HOST"
+	infobloxWAPIUsernameEnvVar  = "INFOBLOX_WAPI_USERNAME"
+	infobloxWAPIPasswordEnvVar  = "INFOBLOX_WAPI_PASSWORD"
+	trustedCAConfigMapEnvVar    = "TRUSTED_CA_CONFIGMAP_NAME"
+	defaultWAPIPort             = "443"
+	defaultWAPIVersion          = "2.3.1"
+	defaultTLSVerify            = "false"
+	defaultHTTPRequestTimeout   = 20
+	defaultHTTPConnPool         = 10
+	defaultGridMasterHostname   = "infoblox.localdomain"
+	defaultHostFilename         = "host"
+	defaultUsernameFilename     = "username"
+	defaultPasswordFilename     = "password"
+	operatorContainerName       = "operator"
+)
+
+type infobloxTestHelper struct {
+	client       *enhancedIBClient
+	gridHost     string
+	wapiUsername string
+	wapiPassword string
+}
+
+var _ providerTestHelper = &infobloxTestHelper{}
+
+func newInfobloxHelper(kubeClient client.Client) (*infobloxTestHelper, error) {
+	helper := &infobloxTestHelper{}
+
+	if err := helper.prepareConfigurations(kubeClient); err != nil {
```

Comment:
:) that one is funny. I passed the client because I knew I'd need it for the TLS configmap injection. But then I forgot to pass it down to the function which does the injection. However there is `kubeClient` global variable too which got used at the end.
Fixed this.